File: backend/models/Reservation.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/db');

const Reservation = sequelize.define('Reservation', {
  id: {
    type: DataTypes.INTEGER,
    autoIncrement: true,
    primaryKey: true,
  },
  bookId: {
    type: DataTypes.INTEGER,
    allowNull: false,
  },
  userId: {
    type: DataTypes.INTEGER,
    allowNull: false,
  },
  status: {
    type: DataTypes.ENUM('pending', 'approved', 'rejected', 'cancelled'),
    allowNull: false,
    defaultValue: 'pending',
  },
  approvedBy: {
    type: DataTypes.INTEGER,
    allowNull: true,
  },
});

module.exports = Reservation;

File: frontend/app/lib/models/reservation.dart
class Reservation {
  final int id;
  final int bookId;
  final int userId;
  final String status;
  final int? approvedBy;

  Reservation({
    required this.id,
    required this.bookId,
    required this.userId,
    required this.status,
    this.approvedBy,
  });

  factory Reservation.fromJson(Map<String, dynamic> json) {
    return Reservation(
      id: json['id'],
      bookId: json['bookId'],
      userId: json['userId'],
      status: json['status'],
      approvedBy: json['approvedBy'],
    );
  }
}
